client.go

```diff
@@ -89,12 +91,16 @@ func NewClient(config *ClientConfig) (*Client, error) {
 type Client struct {
 	username, password, baseURL string
 	client                      *retryablehttp.Client
+	securityPath                string
 }
 
 // Role management
 
 func (c *Client) CreateRole(ctx context.Context, name string, role map[string]interface{}) error {
-	endpoint := "/_xpack/security/role/" + name
+	if err := c.setSecurityPath(ctx); err != nil {
```

Comment:
Instead of calling `setSecurityPath` in each of these methods, can we call in once during NewClient to determine and set `c.securityPath`? It seems that this is only set once since subsequent calls no-ops. 

Since `Client` is not protected by a mutex, checking and modifying `securityPath` through these methods might be racy if they are called concurrently.

Reply:
Yep, good point! I refactored this to call `setSecurityPath` from NewClient in cbe7ef7.